### ISSUE TITLE
Quest links in chat broken

### ISSUE THREAD

**Issue by Isilrond** [Questie/Questie#4408]
Hello,

when i click on any questlinks in the chat i just get an chat outprint like this:

**questieID 13338 Player-4440-037CA96A**

I already deleted all Questie-related data in my WTF folder deleted and reinstalled Questie.

German client and it was posted in the zone chat.

**Comment by BreakBB** [comment]
I'll take a look, thanks for reporting this @Isilrond

**Comment by Isilrond** [comment]
still not working

**Comment by BreakBB** [comment]
Hey @Isilrond so I finally got time to look into this and couldn't find any issue with the quest links.

Are you playing on a Wotlk server or on Classic era? Did you try to disable all other Addons except Questie?

![grafik](https://user-images.githubusercontent.com/33514570/211154950-ea25c913-b3b2-49d0-9f7b-491013e35414.png)

**Comment by Isilrond** [comment]
WotLk

I logged in an alt without any addons except questie and got an error that i have never seen before:
![grafik](https://user-images.githubusercontent.com/42411053/211206627-b2920cb8-3591-4932-b952-9f80eb39f5e5.png)

but the questlinks are working - going to activate the other addons step-by-step now

Getting this LUA Error after i activated "ProjectAzilroka" - the last line is the output i reported in my first post:

Message: ...nterface\AddOns\Tukui\Modules\ChatFrames\CopyURL.lua:54: ItemRefTooltip:SetHyperlink(): Unknown link type
Time: Sun Jan  8 17:08:34 2023
Count: 1
Stack: ...nterface\AddOns\Tukui\Modules\ChatFrames\CopyURL.lua:54: ItemRefTooltip:SetHyperlink(): Unknown link type
[string "=[C]"]: ?
[string "@Interface\AddOns\Tukui\Modules\ChatFrames\CopyURL.lua"]:54: in function `SetHyperlink'
[string "@Interface\FrameXML\ItemRef.lua"]:362: in function <Interface\FrameXML\ItemRef.lua:1>
[string "@Interface_Wrath\AddOns\Blizzard_CombatLog\Blizzard_CombatLog.lua"]:3516: in function `SetItemRef'
[string "@Interface\FrameXML\ChatFrame.lua"]:3763: in function <Interface\FrameXML\ChatFrame.lua:3762>
[string "=[C]"]: in function `ChatFrame_OnHyperlinkShow'
[string "*:OnHyperlinkClick"]:1: in function <[string "*:OnHyperlinkClick"]:1>
[string "=[C]"]: ?

Locals: (*temporary) = ItemRefTooltip {
 0 = <userdata>
 Shadow = <unnamed> {
 }
 layoutType = "TooltipDefaultLayout"
 shoppingTooltips = <table> {
 }
 GetBackdropColor = <function> defined @Interface\SharedXML\SharedTooltipTemplates.lua:191
 SetHyperlink = <function> defined @Interface\AddOns\Tukui\Modules\ChatFrames\CopyURL.lua:40
 BottomOverlay = <unnamed> {
 }
 SetBackdropBorderColor = <function> defined @Interface\SharedXML\SharedTooltipTemplates.lua:195
 updateTooltip = 0.200000
 IsSkinned = true
 TooltipBackdropOnLoad = <function> defined @Interface\SharedXML\SharedTooltipTemplates.lua:172
 TopOverlay = <unnamed> {
 }
 GetBackdropBorderColor = <function> defined @Interface\SharedXML\SharedTooltipTemplates.lua:199
 Backdrop = <unnamed> {
 }
 needsReset = true
 SetBorderBlendMode = <function> defined @Interface\SharedXML\SharedTooltipTemplates.lua:203
 SetBackdropColor = <function> defined @Interface\SharedXML\SharedTooltipTemplates.lua:187
 NineSlice = <unnamed> {
 }
 UpdateTooltip = <function> defined *:OnLoad:5
}
(*temporary) = "questie:1143:Player-4440-02CBD859"

**Comment by BreakBB** [comment]
Hey @Isilrond thanks a lot for coming back to this!

The screenshot you posted shows a rare error of Questie, which should be fixed with a `/reload`.

Hmm... ElvUI is catching the error and propagates it up for you. But it seems one of your addons can't handle the Questie-Links 🤔 If that only happens after you enabled "ProjectAzilroka" (what ever that is 🤷🏻 ), you should try to reach out to them!

**Comment by Isilrond** [comment]
Using TukUI - it's close to ElvUI - and ProjectAzilroka is an addon that only works together with ElvUI/TukUI - and for ProjectAzilroka use only the feature to move the minimap buttons below the minimap - every other feature is disabled.

![grafik](https://user-images.githubusercontent.com/42411053/211208915-defca7a0-17e8-4c70-bc3b-fb4ab0d0725b.png)

no chance i go back to this :-))
![grafik](https://user-images.githubusercontent.com/42411053/211208981-28a18036-e98a-4fb7-abad-d411c2249c0b.png)

**Comment by drejjmit** [comment]
> no chance i go back to this :-)) 
![grafik](https://user-images.githubusercontent.com/42411053/211208981-28a18036-e98a-4fb7-abad-d411c2249c0b.png)

https://www.curseforge.com/wow/addons/mbb-classic
This might be an alternative until they fix the issue 🙈

**Comment by Azilroka** [comment]
`Message: ...nterface\AddOns\Tukui\Modules\ChatFrames\CopyURL.lua:54: ItemRefTooltip:SetHyperlink(): Unknown link type`

Tukui chat copy window which you clicked doesn't understand the link type.

Quest string and [QuestLinks](https://wowpedia.fandom.com/wiki/QuestLink) were added in patch 2.4

I would assume this is broke due to WotLK being regressed in various aspects and patched forward from TBC Classic instead of being properly done. There is alot of regression in the Classic branches as of late.

**Comment by Isilrond** [comment]
just saying it's working with TukUI + Questie active + but when i activate ProjectAzilroka - it no longer works. 
Feels like TukUI dev is anyway superlazy and there is no way not to change to elvUI

**Comment by Azilroka** [comment]
Now try with ProjectAzilroka and Questie.

**Comment by Isilrond** [comment]
Yeah no problem - also works when TukUI active - but chat module disabled....

But i opened already enough "issues" on TukUI github - last update was a month ago. I expect it will be neve fixed anyway. 

Issue closed.

**Comment by Muehe** [comment]
I think we should keep this open until it is clear which combination of Addons causes the problem or if it is a Blizzard bug.

**Comment by Azilroka** [comment]
Pretty simple to do @Muehe. Does quest linking work in the chat frame without Questie? If not.. then using SetHyperlink API on the unmodified ItemRefTooltip will throw an error since it will be an invalid hyperlink,

**Comment by Muehe** [comment]
@Azilroka Not sure I follow, from what I understood Questie links are the ones breaking?

Anyway, we are low on dev time at the moment, and have to focus on the 10.x/3.4.1 API changes, hence just reopening here as to not lose track.

**Comment by Azilroka** [comment]
No, Tukui tries to set the hyperlink in the CopyChat which is what I was saying.. It doesn't understand the questie link which would have to be excluded in Tukui itself. If it was a proper hyperlink that WotLK supported the issue wouldn't be here. That's what I was saying.

It's not a Questie issue or a Blizzard issue.

**Comment by Muehe** [comment]
But it would be avoided entirely by Questie changing the link format to "proper" hyperlinks? Because that might be worthwhile then.

Thanks for all the input btw. :+1:

**Comment by Azilroka** [comment]
Well the SetHyperlink doesn't support Quest linking.. I don't think that was formally introuduced until Cata

**Comment by BreakBB** [comment]
@Muehe did you find more time to look into this?

**Comment by Crumdidlyumshis** [comment]
bump?

**Comment by MyTechnoHunter** [comment]
Invalid/fixed with current version
